### PR TITLE
Add New Board: nRF52840-BBoard

### PR DESCRIPTION
**Pull request by ddB0515**
Adding nRF52840-bboard

More info can be found here
https://github.com/ddB0515/nRF52840-BBoard

### CODE REVIEW

File: .github/workflows/githubci.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ jobs:
           - 'mikoto'
           - 'nice_nano'
           - 'nrf52840_m2'
+          - 'nrf52840_bboard'
```
Comment (hathach) [inline .github/workflows/githubci.yml:46]: should be 1 line up

File: src/boards/nrf52840_bboard/board.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+//--------------------------------------------------------------------+
+#define BLEDIS_MANUFACTURER   "ddB0515"
+#define BLEDIS_MODEL          "nRF52840 BBoard"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID           0x1d50
+#define USB_DESC_UF2_PID       0x6158
+#define USB_DESC_CDC_ONLY_PID  0x6157
```
Comment (hathach) [inline src/boards/nrf52840_bboard/board.h:60]: could you provide the link where the vid is allocated to your board

**Review by hathach — COMMENTED** [review]
thank you for you PR. It looks good, there is only a couple of minor questions.

File: .github/workflows/githubci.yml (hunk 1 above)
Comment (ddB0515) [reply .github/workflows/githubci.yml:46]: no problem will change it

File: src/boards/nrf52840_bboard/board.h (hunk 2 above)
Comment (ddB0515) [reply src/boards/nrf52840_bboard/board.h:60]: here is PR on openmoko
https://github.com/openmoko/openmoko-usb-oui/pull/29

Comment (hathach) [reply src/boards/nrf52840_bboard/board.h:60]: thanks for the link, both USB_DESC_UF2_PID and USB_DESC_CDC_ONLY_PID should be `0x6157`. Both are in bootloader mode, one with msc (uf2) and one without it (cdc only) for fast enumeration. MSC could take 2 more seconds to show up. Though it is safe to share PID between 2 modes.

Comment (ddB0515) [reply src/boards/nrf52840_bboard/board.h:60]: sure thing, I didn't know that but will change it to follow your recommendation and use `0x6157`
Thanks

**Review by hathach — APPROVED** [review]
look great, thank you

**Pull request merged**